sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/action/DataChangeRecordAction.java

```diff
@@ -88,6 +91,14 @@ public Optional<ProcessContinuation> run(
 
     final Timestamp commitTimestamp = record.getCommitTimestamp();
     final Instant commitInstant = new Instant(commitTimestamp.toSqlTimestamp().getTime());
+    if (tracker instanceof Interruptible
```

Comment:
Could we receive an instance of Interruptible instead of having to do an instanceof comparison on every call?

Reply:
Added as explicit class as this route proved to be infeasible due to Beam adapter pattern